src/Servers/Kestrel/Core/src/ListenOptionsHttpsExtensions.cs

```diff
@@ -163,35 +163,22 @@ public static ListenOptions UseHttps(this ListenOptions listenOptions, Action<Ht
     {
         ArgumentNullException.ThrowIfNull(configureOptions);
 
+        // We consider calls to `UseHttps` to be a clear expression of user intent to pull in HTTPS configuration support
+        listenOptions.KestrelServerOptions.EnableHttpsConfiguration();
```

Comment:
Note that I _didn't_ do this for the overloads below that explicitly say they don't consume/respect configuration.

Reply:
Added a test documenting this behavior.